chainstate/src/detail/ban_score.rs

```diff
@@ -0,0 +1,145 @@
+            // this is zero because it's used when we add the outputs whose transactions we tested beforehand
+            StateUpdateError::InvariantErrorTxNumWrongInBlock(_, _) => 0,
+            StateUpdateError::OutputAlreadyPresentInInputsCache => 100,
+            StateUpdateError::ImmatureBlockRewardSpend => 100,
+            StateUpdateError::PreviouslyCachedInputNotFound => 0,
+            StateUpdateError::PreviouslyCachedInputWasErased => 100,
+            StateUpdateError::InvariantBrokenAlreadyUnspent => 0,
+            // Even though this is an invariant error, it stems from referencing a block for reward that doesn't exist
+            StateUpdateError::InvariantBrokenSourceBlockIndexNotFound => 100,
+            StateUpdateError::MissingOutputOrSpent => 100,
+            StateUpdateError::MissingOutputOrSpentOutputErasedOnConnect => 100,
+            StateUpdateError::MissingOutputOrSpentOutputErasedOnDisconnect => 0,
+            StateUpdateError::AttemptToPrintMoney(_, _) => 100,
+            StateUpdateError::TxFeeTotalCalcFailed(_, _) => 100,
+            StateUpdateError::OutputAdditionError => 100,
+            StateUpdateError::SignatureVerificationFailed => 100,
+            StateUpdateError::InvalidOutputCount => 100,
+            StateUpdateError::BlockHeightArithmeticError => 100,
+            StateUpdateError::InputAdditionError => 100,
+            StateUpdateError::DoubleSpendAttempt(_) => 100,
+            StateUpdateError::OutputIndexOutOfRange {
+                tx_id: _,
+                source_output_index: _,
+            } => 100,
+            // Even though this is an invariant error, it stems from a transaction that doesn't exist
+            StateUpdateError::InvariantErrorTransactionCouldNotBeLoaded(_) => 100,
+            // Even though this is an invariant error, it stems from a block reward that doesn't exist
+            StateUpdateError::InvariantErrorHeaderCouldNotBeLoaded(_) => 100,
+            StateUpdateError::FailedToAddAllFeesOfBlock(_) => 100,
+            StateUpdateError::RewardAdditionError(_) => 100,
+            // Even though this is an invariant, we consider it a violation to be overly cautious
+            StateUpdateError::SerializationInvariantError(_) => 100,
+        }
+    }
+}
+
+impl BanScore for CheckBlockError {
+    fn ban_score(&self) -> u32 {
+        match self {
+            CheckBlockError::StorageError(_) => 0,
+            CheckBlockError::MerkleRootMismatch => 100,
+            CheckBlockError::WitnessMerkleRootMismatch => 100,
+            CheckBlockError::BlockConsistencyError(_) => 100,
+            CheckBlockError::InvalidBlockNoPrevBlock => 100,
+            // even though this may be an invariant error, we treat it strictly
+            CheckBlockError::PrevBlockNotFound(_, _) => 100,
+            CheckBlockError::BlockTimeOrderInvalid => 100,
+            CheckBlockError::BlockFromTheFuture => 100,
```

Comment:
This sounds quite harsh, considering the clocks are not going to be perfectly in sync. I think we should agree on an acceptable clock disparity and handle this error more gracefully.

Reply:
The current implementation is incorrect. This should be a few hours into the future for this to be an error.